superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -79,19 +79,15 @@ export const getSlicePayload = (
     adhocFilters = extractAddHocFiltersFromFormData(formDataFromSlice);
   }
 
-  if (
-    isEmpty(adhocFilters?.adhoc_filters) &&
-    isEmpty(formDataFromSlice) &&
-    formDataWithNativeFilters?.adhoc_filters?.[0]?.operator ===
-      Operators.TEMPORAL_RANGE
-  ) {
-    adhocFilters.adhoc_filters = [
-      {
-        ...formDataWithNativeFilters.adhoc_filters[0],
-        comparator: 'No filter',
-      },
-    ];
-  }
+  // This loop iterates through the adhoc_filters array in formDataWithNativeFilters.
+  // If a filter is of type TEMPORAL_RANGE and isExtra, it sets its comparator to
+  // 'No filter' and adds the modified filter to the adhocFilters array. This ensures that all
+  // TEMPORAL_RANGE filters are converted to 'No filter' when saving a chart.
+  formDataWithNativeFilters?.adhoc_filters?.forEach(filter => {
```

Comment:
Previously, `formDataWithNativeFilters.adhoc_filters `were only used if `adhocFilters.adhoc_filters` and `formDataFromSlice` were empty. Why were these checks necessary? Don't we need them anymore?

Reply:
In the past, we used those checks for `adhocFilters.adhoc_filters` and `formDataFromSlice` being empty because we noticed that the `TEMPORAL_RANGE` filter would disappear when those conditions were met. But as we dug deeper, we found out that this wasn't just happening when both of them were empty. It was also happening when `adhocFilters.adhoc_filters` had another filter. This PR is attempting to address both scenarios, ensuring that the TEMPORAL_RANGE filter is retained properly.

Reply:
Thank you for the additional context @lilykuang. Do you know if the fixes in https://github.com/apache/superset/pull/24405 and https://github.com/apache/superset/pull/24876 are not affected by this change?

Reply:
they should be affected

Reply:
Sorry, I'm not sure if I was clear. What I mean is, does your change also preserves the fixes that were made in those PRs? 
- Creating a new chart with dashboard time filters but filter can't apply again 
- Invalid "No Filter" applied

Reply:
This change is okay for my hotfix #24876